compiler/tests-common/tests/org/jetbrains/kotlin/codegen/AbstractCheckLocalVariablesTableTest.kt

```diff
@@ -33,43 +35,86 @@ abstract class AbstractCheckLocalVariablesTableTest : CodegenTestCase() {
         compile(files)
 
         try {
-            val classAndMethod = parseClassAndMethodSignature(wholeFile)
+            val filteredLines = filterTestFileForTargetBackend(wholeFile)
+            val classAndMethod = parseClassAndMethodSignature(filteredLines)
             val split = classAndMethod.split("\\.".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
             assert(split.size == 2) { "Exactly one dot is expected: $classAndMethod" }
             val classFileRegex = StringUtil.escapeToRegexp(split[0] + ".class").replace("\\*", ".+")
             val methodName = split[1]
 
             val outputFiles = (classFileFactory as OutputFileCollection).asList()
-            val outputFile = outputFiles.first { file -> file.relativePath.matches(classFileRegex.toRegex()) }
-
-            val pathsString = outputFiles.joinToString { it.relativePath }
-            assertNotNull("Couldn't find class file for pattern $classFileRegex in: $pathsString", outputFile)
+            val outputFile = outputFiles.firstOrNull { file -> file.relativePath.matches(classFileRegex.toRegex()) }
+            checkNotNull(outputFile) {
+                val pathsString = outputFiles.joinToString { it.relativePath }
+                "Couldn't find class file for pattern $classFileRegex in: $pathsString"
+            }
 
             val classReader = ClassReader(outputFile.asByteArray())
             val actualLocalVariables = readLocalVariable(classReader, methodName)
             checkLocalVariableTypes(classReader, methodName, actualLocalVariables)
 
-            doCompare(wholeFile, files.single().content, actualLocalVariables)
+            doCompare(filteredLines, files.single().content, actualLocalVariables)
         } catch (e: Throwable) {
             printReport(wholeFile)
             throw e
         }
     }
 
-    protected open fun doCompare(
-        testFile: File,
+    // TODO: Refactor test infrastructure to share filtering with AbstractBytecodeTextTest
+    private fun filterTestFileForTargetBackend(testFile: File): List<String> {
+        val filteredLines = mutableListOf<String>()
+        var currentBackend = TargetBackend.ANY
+        for (line in testFile.readLines()) {
+            if (line.contains(JVM_TEMPLATES)) {
+                currentBackend = TargetBackend.JVM
+            } else if (line.contains(JVM_IR_TEMPLATES)) {
+                currentBackend = TargetBackend.JVM_IR
+            } else if (currentBackend == TargetBackend.ANY || currentBackend == backend) {
+                filteredLines.add(line)
+            }
+        }
+        return filteredLines
+    }
+
+    private fun doCompare(
+        testFileLines: List<String>,
         text: String,
```

Comment:
The `text` parameter is not used anymore

Reply:
Fixed